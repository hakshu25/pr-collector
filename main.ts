import { Command } from "cliffy/command/mod.ts";
import { action } from "./action.ts";

export const command = new Command()
  .name("pr-collector")
  .version("1.0.0")
  .description("Collect PRs from multiple GitHub repositories.")
  .usage("[options] [REPOSITORY_ARGS]")
  .arguments("<REPOSITORY_ARGS:string[]>")
  .env("GITHUB_USER_TOKEN=<token:string>", "GitHub Personal Access token.", {
    required: true,
  })
  .action(action);

// Learn more at https://deno.land/manual/examples/module_metadata#concepts
if (import.meta.main) {
  await command.parse(Deno.args);
}
