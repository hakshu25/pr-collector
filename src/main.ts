import { Command, EnumType } from "./deps/deps.ts";
import { action } from "./action.ts";

enum PRState {
  Open = "open",
  Closed = "closed",
}

const PRStateType = new EnumType(PRState);

export const command = new Command()
  .name("pr-collector")
  .version("1.0.0")
  .description("Collect PRs from multiple GitHub repositories.")
  .usage("[options]")
  .env("GITHUB_USER_TOKEN=<token:string>", "GitHub Personal Access token.", {
    required: true,
  })
  .type("state", PRStateType)
  .option(
    "-s --state <state:state>",
    'Filter PRs by state. Default to "open".',
    { default: "open" },
  )
  .option(
    "-u --user <user:string>",
    "GitHub user name. Default to the authenticated user.",
  )
  .option(
    "-l --limit <limit:number>",
    "Limit the number of PRs to fetch. Default to 100.",
    { default: 100 },
  )
  .action(action);

// Learn more at https://deno.land/manual/examples/module_metadata#concepts
if (import.meta.main) {
  await command.parse(Deno.args);
}
