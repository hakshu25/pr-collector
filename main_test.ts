import { snapshotTest } from "https://deno.land/x/cliffy@v1.0.0-rc.3/testing/mod.ts";
import { command } from "./main.ts";

await snapshotTest({
  name: "pr-collector",
  meta: import.meta,
  osSuffix: ["darwin"],
  steps: {
    "should output help with -h flag": { args: ["-h"] },
    "should output help with --help flag": { args: ["--help"] },
    "should output short version with -V flag": { args: ["-V"] },
    "should output long version with --version flag": { args: ["--version"] },
    "should print help message and error message without args": {
      canFail: true,
    },
    "should print error message for unknown option": {
      args: ["--unknown-option"],
      canFail: true,
    },
  },
  async fn() {
    await command.parse();
  },
});
