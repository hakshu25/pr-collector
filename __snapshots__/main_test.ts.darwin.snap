export const snapshot = {};

snapshot[`pr-collector > should output help with -h flag 1`] = `
stdout:
'
Usage:   pr-collector [options]
Version: 1.0.0                 

Description:

  Collect PRs from multiple GitHub repositories.

Options:

  -h, --help              - Show this help.                                                                                  
  -V, --version           - Show the version number for this program.                                                        
  -s, --state    <state>  - Filter PRs by state. Default to "open".               (Default: \\x1b[32m"open"\\x1b[39m, Values: \\x1b[32m"open"\\x1b[39m, \\x1b[32m"closed"\\x1b[39m)
  -u, --user     <user>   - GitHub user name. Default to the authenticated user.                                             

Environment variables:

  GITHUB_USER_TOKEN  <token>  - GitHub Personal Access token.  (required)

'
stderr:
""
`;

snapshot[`pr-collector > should output help with --help flag 1`] = `
stdout:
'
Usage:   pr-collector [options]
Version: 1.0.0                 

Description:

  Collect PRs from multiple GitHub repositories.

Options:

  -h, --help              - Show this help.                                                                                  
  -V, --version           - Show the version number for this program.                                                        
  -s, --state    <state>  - Filter PRs by state. Default to "open".               (Default: \\x1b[32m"open"\\x1b[39m, Values: \\x1b[32m"open"\\x1b[39m, \\x1b[32m"closed"\\x1b[39m)
  -u, --user     <user>   - GitHub user name. Default to the authenticated user.                                             

Environment variables:

  GITHUB_USER_TOKEN  <token>  - GitHub Personal Access token.  (required)

'
stderr:
""
`;

snapshot[`pr-collector > should output short version with -V flag 1`] = `
stdout:
"1.0.0
"
stderr:
""
`;

snapshot[`pr-collector > should output long version with --version flag 1`] = `
stdout:
"pr-collector 1.0.0
"
stderr:
""
`;

snapshot[`pr-collector > should print help message and error message without args 1`] = `
stdout:
'
Usage:   pr-collector [options]
Version: 1.0.0                 

Description:

  Collect PRs from multiple GitHub repositories.

Options:

  -h, --help              - Show this help.                                                                                  
  -V, --version           - Show the version number for this program.                                                        
  -s, --state    <state>  - Filter PRs by state. Default to "open".               (Default: \\x1b[32m"open"\\x1b[39m, Values: \\x1b[32m"open"\\x1b[39m, \\x1b[32m"closed"\\x1b[39m)
  -u, --user     <user>   - GitHub user name. Default to the authenticated user.                                             

Environment variables:

  GITHUB_USER_TOKEN  <token>  - GitHub Personal Access token.  (required)

'
stderr:
'  error: Missing required environment variable "GITHUB_USER_TOKEN".

'
`;

snapshot[`pr-collector > should print error message for unknown option 1`] = `
stdout:
'
Usage:   pr-collector [options]
Version: 1.0.0                 

Description:

  Collect PRs from multiple GitHub repositories.

Options:

  -h, --help              - Show this help.                                                                                  
  -V, --version           - Show the version number for this program.                                                        
  -s, --state    <state>  - Filter PRs by state. Default to "open".               (Default: \\x1b[32m"open"\\x1b[39m, Values: \\x1b[32m"open"\\x1b[39m, \\x1b[32m"closed"\\x1b[39m)
  -u, --user     <user>   - GitHub user name. Default to the authenticated user.                                             

Environment variables:

  GITHUB_USER_TOKEN  <token>  - GitHub Personal Access token.  (required)

'
stderr:
'  error: Missing required environment variable "GITHUB_USER_TOKEN".

'
`;

snapshot[`pr-collector > should print error message for unknown command 1`] = `
stdout:
'
Usage:   pr-collector [options]
Version: 1.0.0                 

Description:

  Collect PRs from multiple GitHub repositories.

Options:

  -h, --help              - Show this help.                                                                                  
  -V, --version           - Show the version number for this program.                                                        
  -s, --state    <state>  - Filter PRs by state. Default to "open".               (Default: \\x1b[32m"open"\\x1b[39m, Values: \\x1b[32m"open"\\x1b[39m, \\x1b[32m"closed"\\x1b[39m)
  -u, --user     <user>   - GitHub user name. Default to the authenticated user.                                             

Environment variables:

  GITHUB_USER_TOKEN  <token>  - GitHub Personal Access token.  (required)

'
stderr:
'  error: Missing required environment variable "GITHUB_USER_TOKEN".

'
`;
